src/test/java/seedu/address/model/room/RoomTest.java

```diff
@@ -0,0 +1,100 @@
+package seedu.address.model.room;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_CAPACITY_DOUBLE;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_ROOM_NUMBER_074;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_TAG_HANDICAP;
+import static seedu.address.testutil.TypicalRooms.DOUBLE_002;
+import static seedu.address.testutil.TypicalRooms.SINGLE_001;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import seedu.address.testutil.RoomBuilder;
+import seedu.address.testutil.TypicalBookings;
+
+public class RoomTest {
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void asObservableList_modifyList_throwsUnsupportedOperationException() {
+        Room room = new RoomBuilder().build();
+        thrown.expect(UnsupportedOperationException.class);
+        room.getTags().remove(0);
+    }
+
+    @Test
+    public void isSameRoom() {
+        // same object -> returns true
+        assertTrue(SINGLE_001.isSameRoom(SINGLE_001));
+
+        // null -> returns false
+        assertFalse(SINGLE_001.isSameRoom(null));
+
+        // different room number -> returns false
+        Room editedSingle001 = new RoomBuilder(SINGLE_001).withRoomNumber("002").build();
+        assertFalse(SINGLE_001.isSameRoom(editedSingle001));
+
+        // same room number, different capacity -> returns true
+        editedSingle001 = new RoomBuilder(SINGLE_001).withCapacity(VALID_CAPACITY_DOUBLE).build();
+        assertTrue(SINGLE_001.isSameRoom(editedSingle001));
+
+        /* KIV Expenses
```

Comment:
Commented blocks should not be in merged code. See Line 86 too.

Reply:
It's a reminder for myself to complete that portion of the test in the future.